File: examples/swift/Sources/ShapeService.swift
import Foundation

/// Provides random shape generation for the example backend.
struct ShapeService: Sendable {
    /// Returns a random integer between 80 and 120 inclusive,
    /// suitable for shape edge lengths.
    static func getRandomShapeEdge() -> Int {
        Int.random(in: 80...120)
    }

    /// Generates a random Shape variant (Circle, Rectangle, or Triangle)
    /// with random edge values between 80 and 120.
    static func generateRandomShape() -> Shape {
        let variant = Int.random(in: 0...2)
        switch variant {
        case 0:
            return .circle(radius: getRandomShapeEdge())
        case 1:
            return .rectangle(width: getRandomShapeEdge(), height: getRandomShapeEdge())
        default:
            return .triangle(
                edgeA: getRandomShapeEdge(),
                edgeB: getRandomShapeEdge(),
                edgeC: getRandomShapeEdge()
            )
        }
    }
}

File: examples/python/src/shape_service.py
import random

from shapes import Circle, Rectangle, Shape, Triangle


def generate_random_shape() -> Shape:
    variant = random.randint(0, 2)
    if variant == 0:
        return Circle(radius=get_random_shape_edge())
    elif variant == 1:
        return Rectangle(width=get_random_shape_edge(), height=get_random_shape_edge())
    else:
        return Triangle(
            edgeA=get_random_shape_edge(),
            edgeB=get_random_shape_edge(),
            edgeC=get_random_shape_edge(),
        )


def get_random_shape_edge() -> int:
    return random.randint(80, 120)
